src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -0,0 +1,336 @@
+        // TODO: Support configurable qos
+        // retry until interrupted
+        topics.forEach(s -> {
+            try {
+                RetryUtils.runWithRetry(mqttExceptionRetryConfig, () -> {
+                    subscribe(s);
+                    // useless return
+                    return null;
+                }, "subscribe-mqtt5-topic", LOGGER);
+                subscribedLocalMqttTopics.add(s);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            } catch (Exception e) {
+                LOGGER.atError().setCause(e).kv(TOPIC, s).log("Failed to subscribe");
+            }
+        });
+    }
+
+    private void resubscribe() {
+        subscribedLocalMqttTopics.clear();
+        // Resubscribe to topics
+        updateSubscriptionsInternal();
+    }
+
+    private void unsubscribe(String topic) {
+        if (client.getIsConnected()) {
+            UnsubscribePacket unsubscribePacket =
+                    new UnsubscribePacket.UnsubscribePacketBuilder().withSubscription(topic).build();
+            LOGGER.atDebug().kv(TOPIC, topic).log("Unsubscribing from MQTT topic");
+            //TODO: deal with unsubscribeFuture
+            CompletableFuture<UnsubAckPacket> unsubscribeFuture = client.unsubscribe(unsubscribePacket);
+        }
+    }
+
+    private synchronized void removeMappingAndSubscriptions() {
+        unsubscribeAll();
+        subscribedLocalMqttTopics.clear();
+    }
+
+    private void unsubscribeAll() {
+        LOGGER.atDebug().kv("mapping", subscribedLocalMqttTopics).log("Unsubscribe from local MQTT topics");
+
+        this.subscribedLocalMqttTopics.forEach(s -> {
+            unsubscribe(s);
+        });
+    }
+
+    @Override
+    public void start() {
+        // TODO: Set callback?
+        try {
+            // client.start();
+            connectAndSubscribe();
+        } catch (CrtRuntimeException e) {
+            // throw new MessageClientException("Unable to start Local Mqtt 5 Client", e);
+        }
+    }
+
+    private synchronized void connectAndSubscribe() throws CrtRuntimeException {
+        LOGGER.atInfo()
+                .kv(BridgeConfig.KEY_BROKER_URI, brokerUri)
+                .kv(BridgeConfig.KEY_CLIENT_ID, clientId)
+                .log("Connecting to broker");
+        reconnectAndResubscribe();
+    }
+
+    private synchronized void doConnect() throws MqttException, KeyStoreException, CrtRuntimeException {
+        if (!client.getIsConnected()) {
+            client.start();
+            LOGGER.atInfo()
+                    .kv(BridgeConfig.KEY_BROKER_URI, brokerUri)
+                    .kv(BridgeConfig.KEY_CLIENT_ID, clientId)
+                    .log("Connected to broker");
+        }
+    }
+
+    private void reconnectAndResubscribe() {
+        int waitBeforeRetry = MIN_WAIT_RETRY_IN_SECONDS;
+
+        while(!client.getIsConnected() && !Thread.currentThread().isInterrupted()) {
```

Comment:
we'll need a different approach than this loop---connect results are in the Mqtt5ClientOptions.LifecycleEvents callback, so at that point we would make the decision whether we can retry the connect.  for now lets just stop if connection fails to keep it simple and add in smarter recovery as a follow-up